packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol

```diff
@@ -485,14 +485,14 @@ contract BridgeFacet is BaseConnextFacet {
         // Enforce liquidity caps
         // NOTE: safe to do this before the swap because canonical domains do
         // not hit the AMMs (local == canonical)
-        if (isCanonical) {
+        uint256 cap = s.caps[key];
+        if (isCanonical && cap > 0) {
           // NOTE: this method includes router liquidity as part of the caps,
           // not only the minted amount
-          uint256 custodied = IERC20(local).balanceOf(address(this)) + _amount;
-          uint256 cap = s.caps[key];
-          if (cap > 0 && custodied > cap) {
+          if (s.custodied[_asset] > cap - _amount) {
             revert BridgeFacet__xcall_capReached();
           }
+          s.custodied[_asset] += _amount;
```

Comment:
this can instead be 
```solidity
uint256 newCustodied = s.custodied[_asset]+_amount;
if (newCustodied > cap) { revert ...; }
s.custodied[_asset] = newCustodied;
```